lib/AST/GenericSignatureBuilder.cpp

```diff
@@ -4401,18 +4401,17 @@ class GenericSignatureBuilder::InferRequirementsWalker : public TypeWalker {
     : module(module), Builder(builder), source(source) { }
 
   Action walkToTypePost(Type ty) override {
-    auto boundGeneric = ty->getAs<BoundGenericType>();
-    if (!boundGeneric)
+    auto decl = ty->getAnyNominal();
+    if (!decl)
       return Action::Continue;
 
-    auto *decl = boundGeneric->getDecl();
     auto genericSig = decl->getGenericSignature();
     if (!genericSig)
-      return Action::Stop;
```

Comment:
Not sure if this is appropriate, @DougGregor?

(Without the `Stop` -> `Continue` switch `Dictionary<T, Int>` would fail to infer `T: Hashable`, because `Int` doesn't have a generic signature.)

Reply:
Oops, yes: this should be Continue so it can continue inferring from other nominal type references in the same type.